ios/StripeTerminalReactNative.swift

```diff
@@ -464,31 +465,36 @@ class StripeTerminalReactNative: RCTEventEmitter, DiscoveryDelegate, BluetoothRe
             if let error = error as NSError? {
                 resolve(Errors.createError(nsError: error))
             } else if let si = si {
-                let setupIntent = Mappers.mapFromSetupIntent(si)
-                self.setupIntents[si.stripeId] = si
+                let uuid = UUID().uuidString
+                let setupIntent = Mappers.mapFromSetupIntent(si,uuid: uuid)
+                self.setupIntents[uuid] = si
                 resolve(["setupIntent": setupIntent])
             }
         }
     }
 
     @objc(collectPaymentMethod:resolver:rejecter:)
     func collectPaymentMethod(params: NSDictionary, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
-        guard let id = params["paymentIntentId"] as? String else {
-            resolve(Errors.createError(code: CommonErrorType.InvalidRequiredParameter, message: "You must provide paymentIntentId."))
+        guard let paymentIntentJSON = params["paymentIntent"] as? NSDictionary else {
+            resolve(Errors.createError(code: CommonErrorType.InvalidRequiredParameter, message: "You must provide paymentIntent."))
+            return
+        }
+        guard let uuid = paymentIntentJSON["sdk_uuid"] as? String else {
+            resolve(Errors.createError(code: CommonErrorType.InvalidRequiredParameter, message: "You must provide sdk_uuid."))
             return
         }
-        guard let paymentIntent = self.paymentIntents[Optional(id)] else {
-            resolve(Errors.createError(code: CommonErrorType.InvalidRequiredParameter, message: "There is no associated paymentIntent with id \(id)"))
+        guard let paymentIntent = self.paymentIntents[uuid] else {
+            resolve(Errors.createError(code: CommonErrorType.InvalidRequiredParameter, message: "There is no associated paymentIntent with uuid \(uuid)"))
```

Comment:
nit-ish: We should probably improve the error messaging here since the UUID is an internal detail that a user wouldn't know how to debug without digging in to the SDK more.

suggestion for if no UUID in the dictionary:
> Unexpected PaymentIntent. collectPaymentMethod requires you to use the PaymentIntent that was returned from either createPaymentIntent or retrievePaymentIntent.

suggestion for if it has a UUID but we don't have a matching entry:
> Unexpected PaymentIntent. The PaymentIntent provided must be re-retrieved with retrievePaymentIntent or a new PaymentIntent must be created with createPaymentIntent. This can happen if the reader disconnected, another PaymentIntent was confirmed, or the PaymentIntent was canceled.

^ got there from reviewing when we clear out self.paymentIntents and disconnect, post confirm, and cancel are the when we do that. So if someone 

1. created a PI A
2. created a PI B
3. collected and confirmed PI A
4. attempted to collect PI B the SDK would error with the miss on paymentIntents[uuid] and they'd get this second error message telling to re-retrieve or create a new PI. (if I understand correctly this shouldn't be a new problem...before they would have gotten the "There is no associated paymentIntent with id \(id)" error)

(and if we do improve the error messages we should do the same in the android layer too)

Reply:
even though it's an internal detail I feel like it is good for users to know exactly what is wrong with the PI they provided. We often get user inquiries on parameters due to vague error messages that don't tell which param or field caused the issue. 

> The PaymentIntent is missing sdk_uuid field.

> No PaymentIntent was found with the sdk_uuid {some_uuid}. The PaymentIntent provided must be re-retrieved with retrievePaymentIntent or a new PaymentIntent must be created with createPaymentIntent.

would you be okay if we changed it to this?

Reply:
yep, sounds good to me